### PR TITLE
AB-4107 Add uc service

### PR DESCRIPTION
**Pull request by Ali-Hajahmed**
- Added create and get credit report end poinnts
 - Added fixtures and tests

AB-4107

### CODE REVIEW

File: src/api/rest/router.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+const schemas = require('../schemas');
+const { errors, getById } = require('../../db-handlers/credit-report');
+const creditReportServiceHandler = require('../../service-handlers/credit-report');
+
+const apiRouter = () => {
+  const router = express.Router();
+  router.use(bodyParser.json());
+  router.use(requestLogger);
+
+  router.post(
```
Comment (tbohwasi) [inline src/api/rest/router.js]: this endpoint does more than just create, no?

File: test/config/index.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+module.exports = {
+  knex: {
+    client: 'mysql',
+    migrations: {
+      directory: `${__dirname}../../../knex/migrations`,
+    },
+    connection: {
+      database: 'exp_user_test',
```
Comment (tbohwasi) [inline test/config/index.js]: we should probably not use the user database

File: src/service-clients/UcClient/errors.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const {
```
Comment (tbohwasi) [inline src/service-clients/UcClient/errors.js]: use consistent folder naming conventions, small caps with dashes.

File: src/service-clients/UcClient/errors.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  UcError,
+  UcMultipleResultsError,
+  UcNoResultsError,
+  UcTemporaryError,
+} = require('@advisa/uc');
+
+module.exports = {
+  UcError,
```
Comment (tbohwasi) [inline src/service-clients/UcClient/errors.js]: is this file necessary, why not use the errors directly from `advisa/uc`?

File: test/spec/api/rest/router.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  });
+
+  describe('_ping', () => {
+    it('responds 200', () => supertest(app.instance())
+      .get('/_ping')
+      .expect(200));
+  });
+
+  describe('POST - /api/v1/credit-reports/create', () => {
+    it('Should fetch credit report and saves it in the database, responds with 200 and the credit report', () => {
```
Comment (tbohwasi) [inline test/spec/api/rest/router.js]: it's more clear here that the name of the endpoint and what it 'should' do are a bit confusing

Comment (tbohwasi) [reply test/spec/api/rest/router.js]: maybe rethink the names or have a more detailed README

File: test/spec/api/rest/router.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+      await supertest(app.instance())
+        .post('/api/v1/credit-reports/create')
+        .send(fixtures.api.rest.postCreditReportCreate.request)
+        .expect(503)
+        .then(({ body }) => {
+          expect(body.message).equal(`UC service: ${fixtures.serviceClients.ucClient.errors.UcTemporaryError.message}`);
+        });
+    });
+
+    it('Should return 400 when uc client returns UcMultipleResultsError', async () => {
```
Comment (tbohwasi) [inline test/spec/api/rest/router.js]: is this the correct error for this case? :thinking:

File: src/api/schemas/credit-report-create-body.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+const joi = require('joi');
+
+module.exports = joi.object().keys({
+  customerId: joi.number().required(),
```
Comment (tbohwasi) [inline src/api/schemas/credit-report-create-body.js]: do we need both customerId and applicantDraftId references?

File: src/api/rest/router.js (hunk 1 above)
Comment (Ali-Hajahmed) [reply src/api/rest/router.js]: Yes! now renamed!

File: test/config/index.js (hunk 2 above)
Comment (Ali-Hajahmed) [reply test/config/index.js]: Fixed

File: .eslintrc [hunk 8]
```diff
@@ -0,0 +1,25 @@
+{
+  "extends": "airbnb-base",
+  "plugins": [
+    "import",
+    "jsdoc"
+  ],
+  "rules": {
+    "no-restricted-syntax": ["error", "ForInStatement", "LabeledStatement", "WithStatement"],
+    "jsdoc/check-tag-names": 1,
+    "jsdoc/check-types": 1,
```
[the remark this answers is not in the document]
Comment (Ali-Hajahmed) [reply .eslintrc]: Done!

File: src/service-clients/UcClient/errors.js (hunk 4 above)
Comment (Ali-Hajahmed) [reply src/service-clients/UcClient/errors.js]: No, I made it so it makes it easy to replace the library in the future

File: src/service-clients/UcClient/errors.js (hunk 3 above)
Comment (Ali-Hajahmed) [reply src/service-clients/UcClient/errors.js]: I made this one to represent the class UcClient
